docusaurus/docs/Android/03-ui/06-guides/09-adding-custom-attachments-v2.mdx

```diff
@@ -0,0 +1,454 @@
+
+```xml
+<merge xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <androidx.appcompat.widget.LinearLayoutCompat
+        android:layout_width="wrap_content"
+        android:layout_height="match_parent"
+        android:gravity="bottom"
+        android:orientation="horizontal"
+        android:paddingHorizontal="4dp"
+        android:paddingVertical="12dp">
+
+        <!-- The new date picker button -->
+        <ImageView
+            android:id="@+id/calendarButton"
+            android:layout_width="32dp"
+            android:layout_height="32dp"
+            android:padding="4dp"
+            android:src="@drawable/ic_calendar"
+            tools:ignore="ContentDescription" />
+
+        <ImageView
+            android:id="@+id/attachmentsButton"
+            android:layout_width="32dp"
+            android:layout_height="32dp"
+            android:padding="4dp"
+            android:src="@drawable/stream_ui_ic_attach"
+            tools:ignore="ContentDescription" />
+
+        <ImageView
+            android:id="@+id/commandsButton"
+            android:layout_width="32dp"
+            android:layout_height="32dp"
+            android:padding="4dp"
+            android:src="@drawable/stream_ui_ic_command"
+            tools:ignore="ContentDescription" />
+
+    </androidx.appcompat.widget.LinearLayoutCompat>
+
+</merge>
+```
+
+```kotlin
+class CustomMessageComposerLeadingContent : FrameLayout, MessageComposerContent {
+
+    private lateinit var binding: CustomMessageComposerLeadingContentBinding
+    private lateinit var style: MessageComposerViewStyle
+
+    var attachmentsButtonClickListener: () -> Unit = {}
+    var commandsButtonClickListener: () -> Unit = {}
+
+    // Click listener for the date picker button
+    var calendarButtonClickListener: () -> Unit = {}
+
+    constructor(context: Context) : this(context, null)
+
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context,
+        attrs,
+        defStyleAttr
+    ) {
+        binding = CustomMessageComposerLeadingContentBinding.inflate(LayoutInflater.from(context), this)
+        binding.attachmentsButton.setOnClickListener { attachmentsButtonClickListener() }
+        binding.commandsButton.setOnClickListener { commandsButtonClickListener() }
+
+        // Set click listener for the date picker button
+        binding.calendarButton.setOnClickListener { calendarButtonClickListener() }
+    }
+
+    override fun attachContext(messageComposerContext: MessageComposerContext) {
+        this.style = messageComposerContext.style
+    }
+
+    override fun renderState(state: MessageComposerState) {
+        // Omitted for the sake of being concise
```

Comment:
I would either add this - even though it's a lot of boilerplate - or specifically note that we can copy paste this part from `DefaultMessageComposerLeadingContent`.

You have already mentioned above that this is a reference, but once again it's nice to be immediately explicit about things to lower the chances of confusing anyone who might have not been paying full attention or misunderstood something.